### PR TITLE
Update stacklevel and add warning for get_output_shape_for

### PR DESCRIPTION
**Pull request by bstriner**
Added `stacklevel=2` in a few places, `stacklevel=3` for preprocessor functions.

Added a warning to `Layer.compute_output_shape`: if a class defines `get_output_shape_for` but is still using the default implementation of `compute_output_shape` from `Layer`, it prints a warning.

This script tests that the warnings all point to the correct lines in the test file (if the stacklevel is set correctly).

This probably isn't everything. The `get_output_shape_for` warning should save some heartache. Does anyone know other hot-spots that need extra warnings?

```python
from keras.models import Model
from keras.layers import Dense, Input, merge, Embedding, Layer, Merge
from keras.regularizers import L1L2
import numpy as np

n=100
_x = np.random.random((n, 10))
_y = np.sum(_x, axis=-1, keepdims=True)

x = Input((_x.shape[1],))
y = Dense(_y.shape[1], W_regularizer=L1L2(1e-6,1e-6))(x)
m = Model(input=[x], output=[y])
m.compile("adam","mean_squared_error")
def gen():
    while True:
        yield _x, _y
m.fit(_x, _y, nb_epoch=5, verbose=0)
m.fit_generator(generator=gen(), samples_per_epoch=32, nb_epoch=5, verbose=0)

embedding = Embedding(50, 100, dropout=0.5)

class L1(Layer):
    def get_output_shape_for(self, s):
        return s
class L2(Layer):
    def dummy(self):
        pass

l1=L1()
l2=L2()
l1.compute_output_shape((10,))
l2.compute_output_shape((10,))

z = Input((_x.shape[1],))
merge_layer = Merge(mode='concat')
merged = merge([x, z], mode="concat")
```

> D:/Projects/keras/keras/legacy_warning_test.py:11: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(1, kernel_regularizer=<keras.reg...)`
>   y = Dense(_y.shape[1], W_regularizer=L1L2(1e-6,1e-6))(x)
> D:/Projects/keras/keras/legacy_warning_test.py:12: UserWarning: Update your `Model` call to the Keras 2 API: `Model(outputs=[Elemwise{..., inputs=[/input_1])`
>   m = Model(input=[x], output=[y])
> D:/Projects/keras/keras/legacy_warning_test.py:17: UserWarning: The `nb_epoch` argument in `fit` has been renamed `epochs`.
>   m.fit(_x, _y, nb_epoch=5, verbose=0)
> D:/Projects/keras/keras/legacy_warning_test.py:18: UserWarning: The semantics of the Keras 2 argument  `steps_per_epoch` is not the same as the Keras 1 argument `samples_per_epoch`. `steps_per_epoch` is the number of batches to draw from the generator at each epoch. Update your method calls accordingly.
>   m.fit_generator(generator=gen(), samples_per_epoch=32, nb_epoch=5, verbose=0)
> D:/Projects/keras/keras/legacy_warning_test.py:18: UserWarning: Update your `fit_generator` call to the Keras 2 API: `fit_generator(verbose=0, generator=<generator..., steps_per_epoch=32, epochs=5)`
>   m.fit_generator(generator=gen(), samples_per_epoch=32, nb_epoch=5, verbose=0)
> D:/Projects/keras/keras/legacy_warning_test.py:20: UserWarning: The `dropout` argument is no longer support in `Embedding`. You can apply a `keras.layers.SpatialDropout1D` layer right after the `Embedding` layer to get the same behavior.
>   embedding = Embedding(50, 100, dropout=0.5)
> D:/Projects/keras/keras/legacy_warning_test.py:31: UserWarning: Class `__main__.L1` defines `get_output_shape_for` but does not override `compute_output_shape`. If this is a Keras 1 layer, please implement `compute_output_shape` to support Keras 2.
>   l1.compute_output_shape((10,))
> D:/Projects/keras/keras/legacy_warning_test.py:35: UserWarning: The `Merge` layer is deprecated and will be removed after 08/2017. Use instead layers from `keras.layers.merge`, e.g. `add`, `concatenate`, etc.
>   merge_layer = Merge(mode='concat')
> D:/Projects/keras/keras/legacy_warning_test.py:36: UserWarning: The `merge` function is deprecated and will be removed after 08/2017. Use instead layers from `keras.layers.merge`, e.g. `sum`, `concatenate`, etc.
>   merged = merge([x, z], mode="concat")

Cheers!

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM, thanks
